pkg/metrics/agent.go

```diff
@@ -47,6 +47,8 @@ type Config struct {
 	Configs                []instance.Config     `yaml:"configs,omitempty,omitempty"`
 	InstanceRestartBackoff time.Duration         `yaml:"instance_restart_backoff,omitempty"`
 	InstanceMode           instance.Mode         `yaml:"instance_mode,omitempty"`
+	DisableKeepAlives      bool                  `yaml:"http_disable_keepalives,omitempty"`
+	IdleConnTimeout        time.Duration         `yaml:"http_idle_conn_timeout,omitempty"`
```

Comment:
IMO these settings make more sense to be exposed in the `metrics.global` block, WDYT?

Reply:
I thought about it, but currently our metrics.global block is something that can be copy/pasted between Prometheus and the Agent and I didn't want to break this symmetry. Do you feel more comfortable putting it there?

Reply:
I'm personally fine with breaking the symmetry, but it's your call

Reply:
My call would be to leave them as-is, except if @mattdurham also has a strong opinion.

Finally, these two options are non-compatible meaning the IdleConnTimeout will only matter for non-disabled Keep-Alive connections. Are we okay with putting a check on the `ApplyDefaults` method?

Reply:
Global means use as default. These settings don't have specific instance settings to apply so I think NOT being in global is reasonable.